### PR TITLE
Lily/debias

### PR DESCRIPTION
**Pull request by lhz1029**
1. finetune/util/debias_configured.py and finetune/util/debias_ops.py are copied from the github.
2. finetune/util/debias_losses.py is copied over with some changes to make it work for sequence labeling (instead of classification) and to except weights (e.g. substituting tf.nn.sparse_softmax_cross_entropy with tf.compat.v1.losses.sparse_softmax_cross_entropy)

### CODE REVIEW

File: finetune/util/debias_models.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class TextBiasModel(BiasModel):
+    def __init__(self, model):
+        self.model = model
+
+    def fit(self, X, y):
+        X = anonymize_headers(X)
+        self.model.fit(X, y)
+    
+    def get_log_probas(self, X):
+        probs = [results[5] for results in self.model.process_long_sequence(X)]
```
Comment (madisonmay) [inline finetune/util/debias_models.py:28]: Hehe, this reminds me that we should probably do something about the return signature of `process_long_sequence`. Logic checks out though.

File: finetune/input_pipeline.py [hunk 2]
```diff
@@ -371,19 +387,40 @@ def get_train_input_fns(self, Xs, Y=None, batch_size=None, val_size=None, contex
             )
         else:
             self._skip_tqdm = 0
-            if context is not None:
-                to_shuffle = (Xs, Y, context)
+            if context is not None and log_proba_biases is not None:
```
Comment (benleetownsend) [inline finetune/input_pipeline.py]: Can we pack everything together into a list of dict or placeholder the tuples with lists of Nones and split on that to reduce the number of code paths?

File: finetune/input_pipeline.py [hunk 3]
```diff
@@ -393,11 +430,13 @@ def get_train_input_fns(self, Xs, Y=None, batch_size=None, val_size=None, contex
                     Xs_tr, Y_tr = dataset_shuffle(*to_shuffle, random_state=self.config.seed)
                     Xs_va, Y_va = self.config.val_set or ([], [])
 
-                Xs_tr, Y_tr, _ = self.resampling(Xs_tr, Y_tr)
-                self.config.dataset_size = len(Xs_tr)
```
Comment (benleetownsend) [inline finetune/input_pipeline.py]: Why can you remove this dataset_size call after resampling?

File: finetune/input_pipeline.py (hunk 2 above)
Comment (lhz1029) [reply finetune/input_pipeline.py]: Sounds good :+1:

**Comment by madisonmay** [comment]
Going to close this as we're discontinuing this line of research for now

**Pull request closed without merging**